File: c/05_sort/selectSort.c
//选择排序
#include <stdio.h>
void selectSort(int arr[],int size){
    int i,j;
    for(i=0;i<size;i++){
        int minIndex=i;
        for(j=i+1;j<size;j++){
            if(arr[j]<arr[minIndex]){
                minIndex=j;
            }
        }
        if(minIndex!=i){
            int tmp=arr[minIndex];
            arr[minIndex]=arr[i];
            arr[i]=tmp;
        }
    }
}

int main(){
    int arr[5]={3,2,1,5,7};
    selectSort(arr,5);
    int i=0;
    for(;i<5;i++){
        printf("%d ",arr[i]);
    }
    printf("\n");
    return 0;
}

File: leetcode/php/cate/sort/selectSort.php
<?php
/**
 * Created by PhpStorm.
 * User: 764432054@qq.com
 * Date: 2020/6/6
 * Time: 22:26
 */

function selectSort(&$arr){
    $len=count($arr);

    for($i=0;$i<$len;$i++){
        $min_index=$i;
        for($j=$i+1;$j<$len;$j++){
            if($arr[$j]<$arr[$min_index]){
                $min_index=$j;
            }
        }
        if($min_index!=$i){
            $tmp=$arr[$min_index];
            $arr[$min_index]=$arr[$i];
            $arr[$i]=$tmp;
        }

    }

}
$arr=[4,2,5,1,6,3,0,9];
selectSort($arr);
print_r($arr);